File: Solutions/1342. Number of Steps to Reduce a Number to Zero/1342. Number of Steps to Reduce a Number to Zero.cpp
class Solution {
public:
    int numberOfSteps (int num) {
        return num ? __builtin_popcount(num) + 31 - __builtin_clz(num) : 0;
    }
};

class Solution {
public:
    int numberOfSteps (int num) {
		if(!num) return 0;
        int res = 0;
        while(num) {
            res += (num & 1) ? 2 : 1;
            num >>= 1;
        }
        return res - 1;
    }
};

File: Solutions/1342. Number of Steps to Reduce a Number to Zero/1342. Number of Steps to Reduce a Number to Zero.go
func numberOfSteps(num int) int {
    if num==0{
        return 0
    }
    res:=0
    for num>0{
        if num&1!=0{
            res+=2
        }else{
            res+=1
        }
        num>>=1
    }
    return res-1
}